### PR TITLE
chore: add cloudantUrl

### PR DESCRIPTION
**Pull request by ChuckCox**
Adding functionality for Kube and Knative toolchain deployment

### CODE REVIEW

**Comment by youming-lin** [comment]
@ChuckCox Have you tested this locally?

**Comment by ChuckCox** [comment]
Tested locally:  
```
npm start

> node-red-app@1.1.1 start /Users/ChuckCox/GitHub/node-red-app
> node --max-old-space-size=160 index.js --settings ./bluemix-settings.js -v

7 Aug 10:56:06 - Starting Node-RED on IBM Cloud bootstrap
7 Aug 10:56:06 - Loading bluemix-settings.js
[2020-08-07T10:56:06.607] [INFO] ibm-cloud-env - Initializing with /Users/ChuckCox/GitHub/node-red-app/config/mappings.json
[2020-08-07T10:56:06.608] [WARN] ibm-cloud-env - File does not exist /Users/ChuckCox/GitHub/node-red-app/config/mappings.json
7 Aug 10:56:06 - Failed to find Cloudant url
7 Aug 10:56:06 - No Cloudant service found
7 Aug 10:56:06 - Falling back to localfilesystem storage. Changes will *not* be saved across application restarts.
7 Aug 10:56:06 - Loading application settings
7 Aug 10:56:06 - Starting first-use setup
7 Aug 10:56:06 - Waiting for first-use setup to complete
7 Aug 10:56:46 - Received first-use setup configuration
7 Aug 10:56:47 - Stopping first-use setup application
7 Aug 10:56:47 - Enabling adminAuth security - set NODE_RED_USERNAME and NODE_RED_PASSWORD to change credentials
7 Aug 10:56:47 - Disabled anonymous read-only access - set NODE_RED_GUEST_ACCESS to 'true' to enable
7 Aug 10:56:47 - [info] 

Welcome to Node-RED
===================

7 Aug 10:56:47 - [info] Node-RED version: v1.1.3-git
7 Aug 10:56:47 - [info] Node.js  version: v11.14.0
7 Aug 10:56:47 - [info] Darwin 19.6.0 x64 LE
7 Aug 10:56:47 - [info] Loading palette nodes
7 Aug 10:56:49 - [info] Settings file  : /Users/ChuckCox/GitHub/node-red-app/bluemix-settings.js
7 Aug 10:56:49 - [info] HTTP Static    : /Users/ChuckCox/GitHub/node-red-app/public
7 Aug 10:56:49 - [info] Context store  : 'default' [module=memory]
7 Aug 10:56:49 - [info] User directory : /Users/ChuckCox/GitHub/node-red-app/.node-red
7 Aug 10:56:49 - [warn] undefined
7 Aug 10:56:49 - [warn] Projects disabled : set editorTheme.projects.enabled=true to enable
7 Aug 10:56:49 - [info] Flows file     : /Users/ChuckCox/GitHub/node-red-app/.node-red/flows.json
7 Aug 10:56:49 - [info] Creating new flow file
7 Aug 10:56:49 - [warn] 

---------------------------------------------------------------------
Your flow credentials file is encrypted using a system-generated key.

If the system-generated key is lost for any reason, your credentials
file will not be recoverable, you will have to delete it and re-enter
your credentials.

You should set your own key using the 'credentialSecret' option in
your settings file. Node-RED will then re-encrypt your credentials
file using your chosen key the next time you deploy a change.
---------------------------------------------------------------------

7 Aug 10:56:49 - [info] Server now running at http://127.0.0.1:1880/red/
7 Aug 10:56:49 - [info] Starting flows
7 Aug 10:56:49 - [info] Started flows
^C7 Aug 10:57:06 - [info] Stopping flows
7 Aug 10:57:06 - [info] Stopped flows
```

**Comment by youming-lin** [comment]
@ChuckCox Have you tested the docker files locally? Are they copied from somewhere? The images don't match between the two.

File: package.json [hunk 1]
```diff
@@ -2,10 +2,11 @@
     "name": "node-red-app",
     "version": "1.1.1",
     "dependencies": {
-        "@cloudant/cloudant": "^4.2.2",
-        "bcrypt": "^3.0.7",
+        "@cloudant/cloudant": "^4.2.4",
+        "bcrypt": "^5.0.0",
         "body-parser": "1.x",
-        "cfenv": "^1.2.2",
+        "cfenv": "^1.2.3",
```
Comment (youming-lin) [inline package.json:8]: We can probably remove this lib. `ibm-cloud-env` replaces this.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
+FROM registry.access.redhat.com/ubi7/ubi
```
Comment (youming-lin) [inline Dockerfile:1]: Image should ideally match between Dockerfile and Dockerfile-tools

The ones in https://github.com/IBM/nodejs-cloudant do

**Comment by ChuckCox** [comment]
I did my testing of the CF deployment, using the "node-red-app-code-monitor" toolchain, and switching the Input value to my ChuckCox-cloudantUrl branch: https://cloud.ibm.com/devops/toolchains/5aafbebd-34a6-4e2d-adb9-4c03d229a463?env_id=ibm:yp:us-south

Here is my toolchain that I did all of my testing of the Helm and Knative deployments in: https://cloud.ibm.com/devops/toolchains/9519c38d-152f-44aa-b170-7c38cadedb70?env_id=ibm:yp:us-south

The "Skit-Pipeline-Helm" and "Skit-Pipeline-Knative" pipelines show failing in the Verification stage only because I did not supply a Slack webhook environment variable for the "slack_message.sh" script (https://github.com/IBM/devex-skit-assets/blob/development/scripts/verification_experience_test.sh#L36), so it exits incorrectly. If you look at the logs for the Verification stage, you will see the message "Skit Experience Test Passed". The Slack webhook will be provided in the newly created "node-red-app-code-monitor" toolchain.

File: bluemix-settings.js [hunk 3]
```diff
@@ -18,8 +18,9 @@ var path = require("path");
 var util = require("util");
 var fs = require("fs");
 
-var cfenv = require("cfenv");
-var appEnv = cfenv.getAppEnv();
+const IBMCloudEnv = require('ibm-cloud-env');
+IBMCloudEnv.init('/server/config/mappings.json');
+var cloudantUrl = IBMCloudEnv.getString('cloudant_url');
```
Comment (jmeis) [inline bluemix-settings.js:23]: Can this be a `const`?

Comment (ChuckCox) [reply bluemix-settings.js:23]: I'm not very proficient with Node.js, so I'm not sure. Maybe @youming-lin or @gee4vee can comment on that.

Comment (ChuckCox) [reply bluemix-settings.js]: I changed `var cloudantUrl` to `const cloudantUrl`

**Review by youming-lin — CHANGES_REQUESTED** [review]
@ChuckCox Can you squash the commits? Otherwise LGTM.

**Pull request merged**